esda/local_join_count_bv.py

```diff
@@ -0,0 +1,225 @@
+import numpy as np
+import pandas as pd
+import warnings
+from scipy import sparse
+from sklearn.base import BaseEstimator
+from libpysal import weights
+from esda.crand import (
+    crand as _crand_plus,
+    njit as _njit,
+    _prepare_univariate,
+    _prepare_bivariate
+)
+
+
+PERMUTATIONS = 999
+
+
+class Local_Join_Count_BV(BaseEstimator):
+
+    """Univariate Local Join Count Statistic"""
+
+    def __init__(self, connectivity=None, permutations=PERMUTATIONS, n_jobs=1, 
+                 keep_simulations=True, seed=None):
+        """
+        Initialize a Local_Join_Count_BV estimator
+        Arguments
+        ---------
+        connectivity     : scipy.sparse matrix object
+                           the connectivity structure describing
+                           the relationships between observed units.
+                           Need not be row-standardized.
+        permutations     : int
+                           number of random permutations for calculation of pseudo
+                           p_values
+        n_jobs           : int
+                           Number of cores to be used in the conditional randomisation. If -1,
+                           all available cores are used.    
+        keep_simulations : Boolean
+                           (default=True)
+                           If True, the entire matrix of replications under the null 
+                           is stored in memory and accessible; otherwise, replications 
+                           are not saved
+        seed             : None/int
+                           Seed to ensure reproducibility of conditional randomizations. 
+                           Must be set here, and not outside of the function, since numba 
+                           does not correctly interpret external seeds 
+                           nor numpy.random.RandomState instances.              
+                           
+        """
+
+        self.connectivity = connectivity
+        self.permutations = permutations
+        self.n_jobs = n_jobs
+        self.keep_simulations = keep_simulations
+        self.seed = seed
+
+    def fit(self, x, z, case="CLC", n_jobs=1, permutations=999):
```

Comment:
Again, generally, sklearn calls these `X` and `y`. Can we use that here?

Reply:
Changed for univariate and bivariate statistics. For multivariate statistics, kept input array as `variables`. Is this okay? Or should we change to `x`? Thought that `variables` is more explicit/clear compared to `x`, but totally open to changing it to be consistent! See https://github.com/pysal/esda/pull/139/commits/2eb0d37d059af4596a2ffd0e103820a7ae0dcd74.